### PR TITLE
Handle projects with many tags better

### PR DESCRIPTION
**Pull request by Nudin**
Implement #46

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -103,6 +103,15 @@ class Release:
     release_type: str
 
 
+@dataclass
+class ReleaseTag:
+    version: str
+    page: str
+    release_type: str
+    tag_url: str = None
```
Comment (konstin) [inline main.py]: If you want `None` as default value (not sure if that's necessary), you need `Optional[str]`

File: main.py [hunk 2]
```diff
@@ -389,23 +398,28 @@ def analyse_tag(
 
     tag_type = release["object"]["type"]
     tag_url = release["object"]["url"]
-    tag_details = get_json_cached(tag_url)
-    if tag_type == "tag":
+    html_url = project_info["html_url"] + "/releases/tag/" + quote_plus(tag_name)
+
+    return ReleaseTag(version=version, page=html_url, release_type=release_type, tag_type=tag_type, tag_url=tag_url)
+
+def get_date_from_tagurl( release: ReleaseTag) -> Release:
```
Comment (konstin) [inline main.py]: Please run `black .` (Black should have been installed with the dev dependencies)

File: main.py [hunk 3]
```diff
@@ -456,18 +470,22 @@ def get_data_from_github(url: str, properties: Dict[str, str]) -> Project:
...
         ]
+        extracted = [v for v in extracted if v is not None]
+        extracted.sort(key=lambda x: LooseVersion(re.sub(r"[^0-9.]", "", x.version)))
+        if len(extracted) > 300:
+            logger.warning(
+                "To many tags ({}), limiting to 300 for performance reasons.".format(
+                    len(extracted)
+                )
+            )
+            extracted = extracted[-100:]
```
Comment (konstin) [inline main.py]: That's a limit of 100 (I'm fine with both 100 and 300; It should only be consistent)

**Comment by konstin** [comment]
I've added some comments, otherwise it's looking good

**Comment by konstin** [comment]
@Nudin I didn't actually run the code yet; Do you want to deploy this now or should I do a manual testing round before updating the bot?

**Comment by Nudin** [comment]
@konstin I've been running it against `puppet` to test it – that worked. I'll might write more improvements tonight.

**Pull request merged**